tools/image-tag

```diff
@@ -10,7 +10,11 @@ fi
 set -o nounset
 
 WIP=$(git diff --quiet || echo '-WIP')
-BRANCH=$(git rev-parse --abbrev-ref HEAD | sed 's#/#-#g')
+
+# In order to be a valid Kubernetes label, this replaces '/' with '-', then replaces
+# any other invalid characters with '.', and removes all non-alphanumeric characters
+# from the beginning of the string. 
+BRANCH=$(git rev-parse --abbrev-ref HEAD | sed 's#/#-#g' | sed 's/[^A-Za-z0-9\._-]/./g' | sed 's/^[^A-Za-z0-9]*//' | head -c51)
```

Comment:
IMO this might be surprising to users. WDYT about having the sanitization in the operator code instead, since that's the only place that needs it?